File: android/app/src/main/java/com/djs66256/short_drama/domain/repository/CommentRepository.kt
package com.djs66256.short_drama.domain.repository

import com.djs66256.short_drama.core.network.ApiResult
import com.djs66256.short_drama.domain.model.Comment
import com.djs66256.short_drama.domain.model.CommentPage
import com.djs66256.short_drama.domain.model.CommentQuery
import com.djs66256.short_drama.domain.model.ToggleCommentLikeResult

interface CommentRepository {
    suspend fun getComments(query: CommentQuery): ApiResult<CommentPage>

    suspend fun createComment(dramaId: String, content: String): ApiResult<Comment>

    suspend fun toggleCommentLike(dramaId: String, commentId: String): ApiResult<ToggleCommentLikeResult>
}

File: ios/ShortDrama/Sources/Data/Repositories/CommentRepository.swift
import Foundation

struct CommentRepository: CommentRepositoryProtocol, Sendable {
    private let dataSource: CommentRemoteDataSource

    init(dataSource: CommentRemoteDataSource = CommentRemoteDataSource()) {
        self.dataSource = dataSource
    }

    func fetchComments(query: CommentQuery) async throws -> PagedResult<Comment> {
        try await dataSource.fetchComments(query: query).toEntity()
    }

    func createComment(dramaId: String, content: String) async throws -> Comment {
        try await dataSource.createComment(dramaId: dramaId, content: content).toEntity()
    }

    func toggleCommentLike(dramaId: String, commentId: String) async throws -> ToggleCommentLikeResult {
        try await dataSource.toggleLike(dramaId: dramaId, commentId: commentId).toEntity()
    }
}
